dot/node_test.go

```diff
@@ -0,0 +1,396 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package dot
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ChainSafe/gossamer/dot/core"
+	"github.com/ChainSafe/gossamer/dot/digest"
+	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/ChainSafe/gossamer/dot/state"
+	dotsync "github.com/ChainSafe/gossamer/dot/sync"
+	"github.com/ChainSafe/gossamer/dot/system"
+	"github.com/ChainSafe/gossamer/dot/telemetry"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/ChainSafe/gossamer/internal/metrics"
+	"github.com/ChainSafe/gossamer/lib/babe"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/ChainSafe/gossamer/lib/grandpa"
+	"github.com/ChainSafe/gossamer/lib/keystore"
+	"github.com/ChainSafe/gossamer/lib/runtime"
+	"github.com/ChainSafe/gossamer/lib/runtime/wasmer"
+	"github.com/ChainSafe/gossamer/lib/services"
+	"github.com/ChainSafe/gossamer/lib/utils"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInitNode(t *testing.T) {
+	cfg := NewTestConfig(t)
+	genFile := newTestGenesisRawFile(t, cfg)
+
+	cfg.Init.Genesis = genFile
+
+	tests := []struct {
+		name   string
+		config *Config
+		err    error
+	}{
+		{
+			name:   "test config",
+			config: cfg,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := InitNode(tt.config)
+			assert.ErrorIs(t, err, tt.err)
+		})
+	}
+}
+
+func TestLoadGlobalNodeName(t *testing.T) {
+	t.Parallel()
+
+	// initialise database using data directory
+	basePath := t.TempDir()
+	db, err := utils.SetupDatabase(basePath, false)
+	require.NoError(t, err)
+
+	basestate := state.NewBaseState(db)
+	basestate.Put(common.NodeNameKey, []byte(`nodeName`))
+
+	err = db.Close()
+	require.NoError(t, err)
+
+	tests := []struct {
+		name         string
+		basepath     string
+		wantNodename string
+		err          error
+	}{
+		{
+			name:         "working example",
+			basepath:     basePath,
+			wantNodename: "nodeName",
+		},
+		{
+			name:     "wrong basepath test",
+			basepath: t.TempDir(),
+			err:      errors.New("Key not found"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotNodename, err := LoadGlobalNodeName(tt.basepath)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+
+			assert.Equal(t, tt.wantNodename, gotNodename)
+		})
+	}
+}
+
+//go:generate mockgen -source=../lib/services/services.go -destination=mock_service_registry_test.go -package=$GOPACKAGE
```

Comment:
Use dis instead (adjust the interfaces you want mocks for at the end)

```suggestion
//go:generate mockgen -destination=mock_service_registry_test.go -package=$GOPACKAGE github.com/ChainSafe/gossamer/lib/services Service,ServiceRegistryIFace
```

Reply:
updated